internal/resources/aws/db_instance.go

```diff
@@ -231,9 +235,70 @@ func (r *DBInstance) BuildResource() *schema.Resource {
 		})
 	}
 
+	if r.PerformanceInsightsEnabled {
+		if r.PerformanceInsightsLongTermRetention {
+			costComponents = append(costComponents, PerformanceInsightsLongTermRetentionCostComponent(r.Region, r.InstanceClass))
+		}
+
+		if r.MonthlyAdditionalPerformanceInsightsRequests == nil || *r.MonthlyAdditionalPerformanceInsightsRequests > 0 {
+			costComponents = append(costComponents,
+				PerformanceInsightsAPIRequestCostComponent(r.Region, r.MonthlyAdditionalPerformanceInsightsRequests))
+		}
+	}
+
 	return &schema.Resource{
 		Name:           r.Address,
 		CostComponents: costComponents,
 		UsageSchema:    DBInstanceUsageSchema,
 	}
 }
+
+func PerformanceInsightsLongTermRetentionCostComponent(region, instanceClass string) *schema.CostComponent {
+	instanceType := strings.TrimPrefix(instanceClass, "db.")
+
+	vCPUCount := decimal.Zero
+	if count, ok := InstanceTypeToVCPU[instanceType]; ok {
+		// We were able to lookup thing VCPU count
+		vCPUCount = decimal.NewFromInt(count)
+	}
+
+	var instanceFamily string
+	split := strings.SplitN(instanceType, ".", 2)
+	if len(split) > 0 {
+		instanceFamily = split[0]
+	}
+
+	return &schema.CostComponent{
+		Name:            fmt.Sprintf("Performance Insights Long Term Retention (%s)", strings.ToLower(instanceClass)),
+		Unit:            "vCPU-month",
+		UnitMultiplier:  decimal.NewFromInt(1),
+		MonthlyQuantity: &vCPUCount,
+		ProductFilter: &schema.ProductFilter{
+			VendorName:    strPtr("aws"),
+			Region:        strPtr(region),
+			Service:       strPtr("AmazonRDS"),
+			ProductFamily: strPtr("Performance Insights"),
+			AttributeFilters: []*schema.AttributeFilter{
+				{Key: "usagetype", ValueRegex: regexPtr("PI_LTR:" + strings.ToUpper(instanceFamily) + "$")},
+			},
+		},
+	}
+}
+
+func PerformanceInsightsAPIRequestCostComponent(region string, additionalRequests *int64) *schema.CostComponent {
```

Comment:
Both resources have access to them within one package, and editor can help navigating to the definition, but I wonder if these two functions should be placed in a dedicated file or something, like `s3_storage_class_helpers.go` does. @hugorut, maybe you know Go's any good practices on the code placement?

Reply:
I think it's fine to have it in this file. I would mark it as un-exported though e.g. `performanceInsightsAPIRequestCostComponent` as it's not consumed outside this package. I think there's only real value to extracting to a dedicated file when there's a ton more specific `s3_storage` helper functions. If there are some lying around already, then maybe we can.